File: application/api/forms/mouse-event.js
/* global Java */

define(['boxing'], function(B) {
    var className = "com.eas.client.forms.events.MouseEvent";
    var javaClass = Java.type(className);
    /**
     * Generated constructor.
     * @constructor MouseEvent MouseEvent
     */
    function MouseEvent() {
        var maxArgs = 0;
        var delegate = arguments.length > maxArgs ?
              arguments[maxArgs] 
            : new javaClass();

        Object.defineProperty(this, "unwrap", {
            configurable: true,
            value: function() {
                return delegate;
            }
        });
        if(MouseEvent.superclass)
            MouseEvent.superclass.constructor.apply(this, arguments);
        delegate.setPublished(this);
        /**
         * Alt key is down on this event.
         */
        this.altDown = true;
        Object.defineProperty(this, "altDown", {
            get: function() {
                var value = delegate.altDown;
                return B.boxAsJs(value);
            }
        });

        /**
         * Which, if any, of the mouse buttons has changed state.
         * Values: 0 - no button, 1 - button 1, 2 - button 2, 3 - button 3.
         */
        this.button = 0;
        Object.defineProperty(this, "button", {
            get: function() {
                var value = delegate.button;
                return B.boxAsJs(value);
            }
        });

        /**
         * The number of mouse clicks associated with this event.
         */
        this.clickCount = 0;
        Object.defineProperty(this, "clickCount", {
            get: function() {
                var value = delegate.clickCount;
                return B.boxAsJs(value);
            }
        });

        /**
         * Ctrl key is down on this event.
         */
        this.controlDown = true;
        Object.defineProperty(this, "controlDown", {
            get: function() {
                var value = delegate.controlDown;
                return B.boxAsJs(value);
            }
        });

        /**
         * Shift key is down on this event.
         */
        this.shiftDown = true;
        Object.defineProperty(this, "shiftDown", {
            get: function() {
                var value = delegate.shiftDown;
                return B.boxAsJs(value);
            }
        });

        /**
         * X cursor coordinate in component's space.
         */
        this.x = 0;
        Object.defineProperty(this, "x", {
            get: function() {
                var value = delegate.x;
                return B.boxAsJs(value);
            }
        });

        /**
         * Meta key is down on this event.
         */
        this.metaDown = true;
        Object.defineProperty(this, "metaDown", {
            get: function() {
                var value = delegate.metaDown;
                return B.boxAsJs(value);
            }
        });

        /**
         * Y cursor coordinate in component's space.
         */
        this.y = 0;
        Object.defineProperty(this, "y", {
            get: function() {
                var value = delegate.y;
                return B.boxAsJs(value);
            }
        });

        /**
         * The source object of the event.
         */
        this.source = new Object();
        Object.defineProperty(this, "source", {
            get: function() {
                var value = delegate.source;
                return B.boxAsJs(value);
            }
        });

    }

    var ScriptsClass = Java.type("com.eas.script.Scripts");
    var space = ScriptsClass.getSpace();
    space.putPublisher(className, function(aDelegate) {
        return new MouseEvent(aDelegate);
    });
    return MouseEvent;
});

File: platypus-js-forms/src/main/java/com/eas/client/forms/events/MouseEvent.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.eas.client.forms.events;

import com.eas.script.NoPublisherException;
import com.eas.script.ScriptFunction;
import com.eas.script.Scripts;
import jdk.nashorn.api.scripting.JSObject;

/**
 *
 * @author mg
 */
public class MouseEvent extends Event<java.awt.event.MouseEvent> {

    protected MouseEvent(java.awt.event.MouseEvent aDelegate) {
        super(aDelegate);
    }

    private static final String X_JS_DOC = ""
            + "/**\n"
            + " * X cursor coordinate in component's space.\n"
            + " */";

    @ScriptFunction(jsDoc = X_JS_DOC)
    public int getX() {
        return delegate.getX();
    }

    private static final String Y_JS_DOC = ""
            + "/**\n"
            + " * Y cursor coordinate in component's space.\n"
            + " */";

    @ScriptFunction(jsDoc = Y_JS_DOC)
    public int getY() {
        return delegate.getY();
    }

    private static final String BUTTON_JS_DOC = ""
            + "/**\n"
            + " * Which, if any, of the mouse buttons has changed state.\n"
            + " * Values: 0 - no button, 1 - button 1, 2 - button 2, 3 - button 3.\n"
            + " */";

    @ScriptFunction(jsDoc = BUTTON_JS_DOC)
    public int getButton() {
        return delegate.getButton();
    }

    private static final String CLICK_COUNT_JS_DOC = ""
            + "/**\n"
            + " * The number of mouse clicks associated with this event.\n"
            + " */";

    @ScriptFunction(jsDoc = CLICK_COUNT_JS_DOC)
    public int getClickCount() {
        return delegate.getClickCount();
    }

    private static final String ALT_DOWN_JS_DOC = ""
            + "/**\n"
            + " * Alt key is down on this event.\n"
            + " */";

    @ScriptFunction(jsDoc = ALT_DOWN_JS_DOC)
    public boolean isAltDown() {
        return delegate.isAltDown() || delegate.isAltGraphDown();
    }

    private static final String CONTROL_DOWN_JS_DOC = ""
            + "/**\n"
            + " * Ctrl key is down on this event.\n"
            + " */";

    @ScriptFunction(jsDoc = CONTROL_DOWN_JS_DOC)
    public boolean isControlDown() {
        return delegate.isControlDown();
    }

    private static final String SHIFT_DOWN_JS_DOC = ""
            + "/**\n"
            + " * Shift key is down on this event.\n"
            + " */";

    @ScriptFunction(jsDoc = SHIFT_DOWN_JS_DOC)
    public boolean isShiftDown() {
        return delegate.isShiftDown();
    }

    private static final String META_DOWN_JS_DOC = ""
            + "/**\n"
            + " * Meta key is down on this event.\n"
            + " */";

    @ScriptFunction(jsDoc = META_DOWN_JS_DOC)
    public boolean isMetaDown() {
        return delegate.isMetaDown();
    }

    @Override
    public JSObject getPublished() {
        if (published == null) {
            JSObject publisher = Scripts.getSpace().getPublisher(this.getClass().getName());
            if (publisher == null || !publisher.isFunction()) {
                throw new NoPublisherException();
            }
            published = (JSObject) publisher.call(null, new Object[]{this});
        }
        return published;
    }
}
